addons/gdscript_sections/gdscript_sections.gd

```diff
@@ -409,8 +411,30 @@ func _on_Dialog_section_added(text: String) -> void:
 	)
 
 	_update_ui_and_display()
+	_section_filter_ui_reset()
 	return
 
+## filters the current ui
+func _on_Dialog_section_filter(text: String) -> void:
+	_update_OverlayDisplay()
+	
+	var sections: Array = data_helper.get_sections(
+		script_editor.get_current_script().get_path(),
+		true
+	)
+
+	tree.clear()
+	var root := tree.create_item()
+
+	for section in sections:
+		if section.text.to_lower().begins_with(text.to_lower()):
```

Comment:
This filter logic isn't satisfactory.

We should replicate the logic used by Godot engine (e.g. the field `Filter by name...` in Input Map). 
This can be easily achieved with RegEx. I can work on this when I have the time.